spot_rl_experiments/spot_rl/utils/skill_executor.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import argparse
+import os
+import os.path as osp
+import time
+
+import rospy
+from spot_rl.envs.skill_manager import SpotSkillManager
+from spot_rl.utils.utils import get_skill_name_and_input_from_ros
+from spot_rl.utils.utils import ros_topics as rt
+
+
+class SpotRosSkillExecutor:
+    """This class reads the ros butter to execute skills"""
+
+    def __init__(self, spotskillmanager):
+        self.spotskillmanager = spotskillmanager
+        self._cur_skill_name_input = None
+
+    def reset_skill_msg(self):
+        """Reset the skill message. The format is skill name, success flag, and message string.
+        This is useful for returning the message (e.g., if skill fails or not) from spot-sim2real to high-level planner.
+        """
+        rospy.set_param("/skill_name_suc_msg", "None,None,None")
+
+    def reset_skill_name_input(self, skill_name, succeded, msg):
+        """Reset skill name and input, and publish the message"""
+        rospy.set_param("/skill_name_input", "None,None")
+        rospy.set_param("/skill_name_suc_msg", f"{skill_name},{succeded},{msg}")
+
+    def execute_skills(self):
+        """Execute skills."""
+
+        # Get the current skill name
+        skill_name, skill_input = get_skill_name_and_input_from_ros()
+        print(f"current skill_name {skill_name} skill_input {skill_input}")
+
+        # Select the skill from the ros buffer and call the skill
+        if skill_name == "Navigate":
```

Comment:
I think its a little weird that this is called every step. Is this resetting the skills etc as a result? I think a better flow would be:

# Alternative code

        # Get the current skill name
        skill_name, skill_input = get_skill_name_and_input_from_ros()
        print(f"current skill_name {skill_name} skill_input {skill_input}")
        
       # If prev skill/ skill input is different than current
       # Execute the code below

Reply:
Hi Akshara, thanks for the discussion. Notice that the skill execution is a blocking call in `skill_executor.py`, which means that if we execute the skill, this program will wait for the skill to finish. The termination of skills is designed inside spotskillmanager, not here. As a result, `skill_executor.py` is actully blocking call, but with the termination in spotskillmanager, the overall behavior is a blocking call. I'm happy to discuss and iterate! Thanks a lot!